File: 프로그래머스/0/181887. 홀수 vs 짝수/홀수 vs 짝수.js
function solution(num_list) {
    return Math.max(
        num_list.filter((_, i) => i % 2 === 0).reduce((a, b) => a + b, 0),
        num_list.filter((_, i) => i % 2 === 1).reduce((a, b) => a + b, 0)
    );
}

File: 프로그래머스/0/181887. 홀수 vs 짝수/홀수 vs 짝수.java
class Solution {
    public int solution(int[] num_list) {  // int[]로 변경
        int oddSum = 0;   // 홀수 번째(1, 3, 5...) 합
        int evenSum = 0;  // 짝수 번째(2, 4, 6...) 합
        
        for (int i = 0; i < num_list.length; i++) {  // .size() → .length
            if (i % 2 == 0) {
                oddSum += num_list[i];  // .get(i) → []
            } else {
                evenSum += num_list[i];
            }
        }
        
        return Math.max(oddSum, evenSum);
    }
}
